### PR TITLE
Hatch subsystem

### PR DESCRIPTION
**Pull request by kchenweijie**

### CODE REVIEW

File: src/main/java/frc/robot/Hatch.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package frc.robot;
+
+import edu.wpi.first.wpilibj.Talon;
+
+public class Hatch {
+    /**
```
Comment (kchenweijie) [inline src/main/java/frc/robot/Hatch.java]: Delete the TODO comment

File: src/main/java/frc/robot/SheetOfKnowledge.java [hunk 2]
```diff
@@ -1,10 +1,16 @@
 package frc.robot;
 
 public class SheetOfKnowledge {
+    /**
```
Comment (kchenweijie) [inline src/main/java/frc/robot/SheetOfKnowledge.java]: Delete the TODO comment

File: src/main/java/frc/robot/Robot.java [hunk 3]
```diff
@@ -17,6 +17,18 @@
  * project.
  */
 public class Robot extends TimedRobot {
+  /**
```
Comment (kchenweijie) [inline src/main/java/frc/robot/Robot.java]: You didn't do anything in this file...

Comment (kchenweijie) [reply src/main/java/frc/robot/Robot.java]: Also don't forget to add code to the OperatorInterface.java file

File: src/main/java/frc/robot/OperatorInterface.java [hunk 4]
```diff
@@ -19,11 +20,24 @@ public OperatorInterface() {
     public void init() {
         this._driveCmd = 0.0;
         this._turnCmd = 0.0;
+        this._hatchCmd = 0.0;
+
```
Comment (kchenweijie) [inline src/main/java/frc/robot/OperatorInterface.java]: Get rid of this extra line

File: src/main/java/frc/robot/OperatorInterface.java [hunk 5]
```diff
@@ -19,11 +20,24 @@ public OperatorInterface() {
...
     }
 
     public void update() {
         this._driveCmd = this._pilotStick.getRawAxis(SheetOfKnowledge.AXIS_DRIVE);
         this._turnCmd = this._pilotStick.getRawAxis(SheetOfKnowledge.AXIS_TURN);
+
+        boolean hatchLift = this._copilotStick.getRawButton(SheetOfKnowledge.BTN_COPILOT_LIFT);
+        double hatchLower = this._copilotStick.getRawAxis(SheetOfKnowledge.AXIS_COPILOT_LOWER);
+        if(hatchLift == true){
```
Comment (kchenweijie) [inline src/main/java/frc/robot/OperatorInterface.java]: Should be if hatchLift is true and hatchLower < 0.5

File: src/main/java/frc/robot/OperatorInterface.java [hunk 6]
```diff
@@ -19,11 +20,24 @@ public OperatorInterface() {
...
+            _hatchCmd = 1.0;
+        }else if(hatchLower >= 0.5){
```
Comment (kchenweijie) [inline src/main/java/frc/robot/OperatorInterface.java]: Sould be if hatchLower >= 0.5 and hatchLift is false

**Review by ghost — APPROVED** [review]

**Pull request merged**